cli/azd/pkg/project/project_manager.go

```diff
@@ -0,0 +1,195 @@
+package project
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/azure/azure-dev/cli/azd/internal/telemetry"
+	"github.com/azure/azure-dev/cli/azd/internal/telemetry/fields"
+	"github.com/azure/azure-dev/cli/azd/pkg/account"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment/azdcontext"
+	"github.com/azure/azure-dev/cli/azd/pkg/exec"
+	"github.com/azure/azure-dev/cli/azd/pkg/ext"
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/azure/azure-dev/cli/azd/pkg/osutil"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools/azcli"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	//nolint:lll
+	projectSchemaAnnotation = "# yaml-language-server: $schema=https://raw.githubusercontent.com/Azure/azure-dev/main/schemas/v1.0/azure.yaml.json"
+
+	ProjectEventDeploy    ext.Event = "deploy"
+	ProjectEventProvision ext.Event = "provision"
+)
+
+var (
+	ProjectEvents []ext.Event = []ext.Event{
+		ProjectEventProvision,
+		ProjectEventDeploy,
+	}
+)
+
+type ProjectManager interface {
```

Comment:
I considered that but since the package exposes both a `ProjectManager` and `ServiceManager` I thought it would be more clear to give `ProjectManager` the prefix.